faculty/clients/environment.py

```diff
@@ -62,18 +63,49 @@ class Constraint(Enum):
     "EnvironmentCreateUpdate", ["name", "description", "specification"]
 )
 
-VERSION_REGEX = re.compile(
+PYTHON_VERSION_REGEX = re.compile(
     r"^(?:\d+\!)?\d+(?:\.\d+)*(?:(?:a|b|rc)\d+)?(?:\.post\d+)?(?:\.dev\d+)?$"
 )
 
+APT_VERSION_REGEX = re.compile(r"^[a-zA-Z0-9\\.\\+-:~]+$")
 
-class VersionSchema(BaseSchema):
+
+class PythonVersionSchema(BaseSchema):
+    constraint = EnumField(Constraint, by_value=True, required=True)
+    identifier = fields.String(required=True)
+
+    @validates("identifier")
+    def validate_version_format(self, data):
+        if not PYTHON_VERSION_REGEX.match(data):
+            raise ValidationError("Invalid version format")
+
+    @post_load
+    def make_version(self, data):
+        return Version(**data)
+
+    @post_dump
+    def dump_version(self, data):
+        self.validate_version_format(data["identifier"])
+        return data
+
+
+class VersionSchema(PythonVersionSchema):
+    def __init__(self, *args, **kwargs):
+        template = (
```

Comment:
Would happliy also just remove this, as I think it's extremely unlikely anyone will be depending on the `VersionSchema` directly.

If it stays, suggest:
```suggestion
        message = (
```